src/main.cc

```diff
@@ -32,11 +31,11 @@ int main(int argc, char** argv) {
     // Create an IO object
     auto io = std::make_shared<mpm::IO>(argc, argv);
 
-    // Set TBB threads
-    unsigned nthreads = tbb::task_scheduler_init::default_num_threads();
-    // If number of TBB threads are positive set to nthreads
-    if (io->nthreads() > 0) nthreads = io->nthreads();
-    tbb::task_scheduler_init init(nthreads);
+    // If number of threads are positive set to nthreads
+    unsigned nthreads = io->nthreads();
+#ifdef _OPENMP
+    omp_set_num_threads(nthreads > 0 ? nthreads : omp_get_max_threads());
+#endif
```

Comment:
Is `io->nthreads()` equal to `XX` set by `export OMP_NUM_THREADS=XX` ?

Reply:
if number of threads are not specified, we use the maximum number of threads. If specified using `-p` or `--parallel` we use those instead.

Reply:
But can one still set the thread by using `OMP_NUM_THREADS`?

Reply:
Yes, that will affect the `omp_get_max_threads()` parameter and will have the same effect as `-p`

Reply:
Understood! Thanks for the clarification